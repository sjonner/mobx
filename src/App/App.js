import React, { Component } from 'react';
import Counter from '../Counter/Counter';
import Todo from '../Todo/Todo';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        {this.props.children}

        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <div className="App-intro">
          <Counter />
          <hr />
          <Todo />
        </div>
      </div>
    );
  }
}

export default App;
