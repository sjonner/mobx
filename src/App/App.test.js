import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'mobx-react';
import renderer from 'react-test-renderer';
import App from './App';
import store from '../stores';

it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
		<Provider store={store}>
			<App />
		</Provider>,
  	div
	);
});

it('renders a snapshot', () => {
  const tree = renderer.create(
  	<Provider store={store}>
			<App />
		</Provider>,
	).toJSON();

  expect(tree).toMatchSnapshot();
});
