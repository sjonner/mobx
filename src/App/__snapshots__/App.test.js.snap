// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <div
    className="App-intro"
  >
    <div>
      Count: 
      0
       times!
      <br />
      <button
        onClick={[Function]}
      >
        +
      </button>
      <button
        onClick={[Function]}
      >
        -
      </button>
      <button
        onClick={[Function]}
      >
        reset
      </button>
    </div>
    <hr />
    <form
      method="POST"
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        name="newTodo"
        placeholder="add a new todo"
      />
      <button
        type="submit"
      >
        add
      </button>
      <div>
        <small>
          <em>
            double click a todo to edit
          </em>
        </small>
        <br />
        <small>
          <em>
            alt + click a todo to mark as done
          </em>
        </small>
        <br />
      </div>
      &lt;none&gt;
    </form>
  </div>
</div>
`;
